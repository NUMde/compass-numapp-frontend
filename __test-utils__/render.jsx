/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';

import UserReducer from '../src/store/user.slice';
import GlobalsReducer from '../src/store/globals.slice';
import QuestionnaireReducer from '../src/store/questionnaire.slice';

const testReducer = combineReducers({
  User: UserReducer,
  Globals: GlobalsReducer,
  Questionnaire: QuestionnaireReducer,
});

function renderWithRedux(
  ui,
  {
    initialState = {
      User: UserReducer.initialState,
      Globals: GlobalsReducer.initialState,
      Questionnaire: QuestionnaireReducer.initialState,
    },
    store = createStore(testReducer, initialState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react-native';

export { renderWithRedux };
