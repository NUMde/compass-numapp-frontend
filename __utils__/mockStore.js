// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

/***********************************************************************************************
imports
***********************************************************************************************/

// redux
/*-----------------------------------------------------------------------------------*/

import thunk from 'redux-thunk';
import { combineReducers, createStore, applyMiddleware } from 'redux';

// reducers
/*-----------------------------------------------------------------------------------*/

import LoginReducer from '../src/screens/login/loginReducer';
import AboutReducer from '../src/screens/about/aboutReducer';
import CheckInReducer from '../src/screens/checkIn/checkInReducer';

// services
/*-----------------------------------------------------------------------------------*/

import localStorage from '../src/services/localStorage';

/***********************************************************************************************
reducer
***********************************************************************************************/

const appReducer = combineReducers({
  Login: LoginReducer,
  CheckIn: CheckInReducer,
  About: AboutReducer,
});

const rootReducer = (state, action) => {
  let localState = { ...state };

  // global actions that need to be available anywhere,
  // will be filtered out and executed by the rootReducer
  // (before the actual reducer is)

  if (action.type === 'USER_LOGOUT') {
    localState = undefined;
    localStorage.removeLastSubjectId();
  }

  if (action.type === 'DELETE_ALL_LOCAL_DATA') {
    localStorage.clearAll();
  }

  if (action.type === 'DELETE_LOCAL_QUESTIONNAIRE') {
    localState.CheckIn.currentCategoryIndex = null;
    localState.CheckIn.questionnaireItemMap = null;
    localState.CheckIn.categories = null;
    localState.CheckIn.categoriesLoaded = false;
    localState.CheckIn.currentPageIndex = 1;

    const { subjectId } = state.CheckIn.user;
    localStorage.removeLastQuestionnaireId(subjectId);
    localStorage.removeCategories(subjectId);
    localStorage.removeQuestionnaireItemMap(subjectId);
  }

  return appReducer(localState, action);
};

/***********************************************************************************************
middleware
***********************************************************************************************/

const middleware = [];
middleware.push(thunk);

/***********************************************************************************************
export
***********************************************************************************************/

let globalMockStore;

export function buildStore(state) {
  if (state) {
    globalMockStore = createStore(
      rootReducer,
      state,
      applyMiddleware(...middleware),
    );
  }
  if (!state) {
    globalMockStore = createStore(rootReducer, applyMiddleware(...middleware));
  }

  return globalMockStore;
}

export default buildStore();
