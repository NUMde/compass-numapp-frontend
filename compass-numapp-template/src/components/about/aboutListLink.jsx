// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

/***********************************************************************************************
imports
***********************************************************************************************/

import React from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { navigationPropType, webViewPropType } from '~propTypes';

// components
import { ListItem } from 'react-native-elements';

// config
import { theme } from '~config';
import { Routes } from '~navigation/constants';

/***********************************************************************************************
 * component:
 * Renders a ListItem which will navigate to another screen when clicked on
 *
 * @param  {object}	props
 * @param  {object} props.navigation the navigation object provided by 'react-navigation'
 * @param  {{
 *			title:string,
 *			subTitle: string,
 *      iconType: string,
 *      iconTitle: string,
 *			uri: string
 * }}	props.webView holds the strings and the link to open in the webView
 ***********************************************************************************************/
function AboutListLink({ navigation, webView }) {
  return (
    <ListItem
      containerStyle={localStyle.containerStyle}
      onPress={() => {
        navigation.navigate(Routes.WEBVIEW, { ...webView });
      }}
      testID="aboutListLink"
    >
      {/* title & subtitle of the listItem - the strings a identified by the webView*/}
      <ListItem.Content>
        <ListItem.Title style={localStyle.title} testID="ALL_title">
          {webView.title}
        </ListItem.Title>

        {webView.subTitle && (
          <ListItem.Subtitle style={localStyle.subTitle} testID="ALL_subTitle">
            {webView.subTitle}
          </ListItem.Subtitle>
        )}
      </ListItem.Content>

      {/* the icon on the right-hand-side */}
      <ListItem.Chevron
        {...{
          type: webView.iconType,
          name: webView.iconTitle,
          color: theme.values.defaultListLinkIconColor,
          reverse: true,
          size: 12,
        }}
      />
    </ListItem>
  );
}

AboutListLink.propTypes = {
  navigation: PropTypes.shape(navigationPropType).isRequired,
  webView: PropTypes.shape(webViewPropType).isRequired,
};

/***********************************************************************************************
local styling
***********************************************************************************************/

const localStyle = StyleSheet.create({
  containerStyle: {
    width: '100%',
    borderBottomColor: theme.colors.accent3,
    borderBottomWidth: 1,
    backgroundColor: theme.values.defaultListLinkBackgroundColor,
    padding: 15,
  },

  subTitle: {
    color: theme.colors.accent4,
    ...theme.fonts.body,
  },

  title: {
    ...theme.fonts.title2,
  },
});

/***********************************************************************************************
export
***********************************************************************************************/

export default AboutListLink;
