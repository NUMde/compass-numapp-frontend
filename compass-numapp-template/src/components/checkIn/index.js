// components for the 'checkin' screen
export { default as CheckInListView } from './checkInListView';
export { default as CheckInTiles } from './checkInTiles';
export { default as CheckInWelcomeText } from './welcomeText';
