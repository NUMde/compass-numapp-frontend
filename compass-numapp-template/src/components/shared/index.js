// ScrollIndicatorWrapper component
export { default as ScrollIndicatorWrapper } from './scrollIndicatorWrapper';

// Spinner component for loading animation
export { default as Spinner } from './spinner';

// Banner Component
export { default as Banner } from './banner';
