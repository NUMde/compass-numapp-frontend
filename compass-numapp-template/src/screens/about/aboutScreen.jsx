// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

/***********************************************************************************************
imports
***********************************************************************************************/

import React, { PureComponent } from 'react';
import { View, TouchableOpacity, Text, StyleSheet } from 'react-native';

import { ListItem } from 'react-native-elements';
import RedirectModal from '../../components/modal/redirectModal';
import Banner from '../../components/banner/banner';
import AboutListItem from '../../components/about/aboutListItem';
import AboutListLink from '../../components/about/aboutListLink';
import ScrollIndicatorWrapper from '../../components/scrollIndicatorWrapper/scrollIndicatorWrapper';

import { appConfig, theme, strings } from '../../config';

let localStyle;

/***********************************************************************************************
Component
***********************************************************************************************/

class AboutScreen extends PureComponent {
  /**
   * renders the About-Screen
   * @constructor
   * @param  {object}    props
   * @param  {object}    props.navigation the navigation object provided by 'react-navigation'
   * @param  {function}  props.logout logs out the user
   * @param  {function}  props.clearAll deletes all local data
   */

  // rendering
  /*-----------------------------------------------------------------------------------*/

  render() {
    const { navigation, actions, clearAll, logout, showModal, modalLink } =
      this.props;
    return (
      <View style={localStyle.wrapper}>
        {/* top banner */}
        <Banner
          nav={navigation}
          title={strings.about.title}
          subTitle={strings.about.subTitle}
          noMenu
        />

        {/* the modal to be opened */}
        <RedirectModal
          showModal={showModal}
          modalLink={modalLink}
          actions={actions}
        />

        {/* ScrollView with content */}
        <View style={localStyle.wrapper}>
          <ScrollIndicatorWrapper
            contentData={
              <View style={localStyle.wrapper}>
                {/* holds the list-items */}
                <View style={localStyle.wrapper}>
                  {/* links to the LegalInformationScreen */}
                  {appConfig.allowAccessToLegalInformationScreen && (
                    <ListItem
                      containerStyle={localStyle.containerStyle}
                      onPress={() => navigation.navigate('LegalInformation')}
                    >
                      {/* title & subtitle of the listItem - the strings a identified by the webView*/}
                      <ListItem.Content>
                        <ListItem.Title style={localStyle.title}>
                          {strings.about.legal.title}
                        </ListItem.Title>

                        <ListItem.Subtitle style={localStyle.subTitle}>
                          {strings.about.legal.subTitle}
                        </ListItem.Subtitle>
                      </ListItem.Content>

                      {/* the icon on the right-hand-side */}
                      <ListItem.Chevron
                        {...{
                          type: strings.about.legal.iconType,
                          name: strings.about.legal.iconTitle,
                          color: theme.values.legalListLinkIconColor,
                          reverse: true,
                          size: 12,
                        }}
                      />

                      {/* the icon on the right-hand-side */}
                    </ListItem>
                  )}

                  {/* iterates over all items in strings.wevViews */}
                  {strings.webViews.map((webView) => (
                    // navigates to the webview screen
                    <AboutListLink
                      key={webView.title}
                      // {...this.props}
                      actions={actions}
                      navigation={navigation}
                      webView={webView}
                    />
                  ))}

                  {/* iterates over all items in strings.modalLinks */}
                  {strings.modalLinks.map((_modalLink) => (
                    // navigates to the webview screen
                    <AboutListItem
                      actions={actions}
                      key={_modalLink.title}
                      modalLink={_modalLink}
                    />
                  ))}
                </View>

                {/* optional buttons on the bottom of the screen - JUST FOR DEVELOPMENT*/}
                <View style={localStyle.bottom}>
                  {/* logout button */}
                  {appConfig.showLogout && (
                    <TouchableOpacity
                      style={localStyle.button}
                      onPress={logout}
                      accessibilityLabel={strings.about.logout}
                      accessibilityRole={strings.accessibility.types.button}
                      accessibilityHint={strings.accessibility.logoutHint}
                    >
                      <Text style={localStyle.buttonLabel}>
                        {strings.about.logout}
                      </Text>
                    </TouchableOpacity>
                  )}

                  {/* delete-all-data button */}
                  {appConfig.showEraseAll && (
                    <TouchableOpacity
                      style={localStyle.buttonAlert}
                      onPress={clearAll}
                      accessibilityLabel={strings.about.delete}
                      accessibilityRole={strings.accessibility.types.button}
                      accessibilityHint={strings.accessibility.logoutHint}
                    >
                      <Text style={localStyle.buttonLabel}>
                        {strings.about.delete}
                      </Text>
                    </TouchableOpacity>
                  )}
                </View>
              </View>
            }
          />
        </View>
      </View>
    );
  }
}

/***********************************************************************************************
local styling
***********************************************************************************************/

localStyle = StyleSheet.create({
  wrapper: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
    width: '100%',
    height: '100%',
    backgroundColor: theme.values.defaultBackgroundColor,
  },

  bottom: {
    flex: 1,
    justifyContent: 'flex-end',
    marginBottom: 36,
    height: '100%',
    marginTop: 20,
    width: '80%',
  },

  button: {
    ...theme.classes.buttonPrimary,
    bottom: 0,
    marginTop: 10,
  },

  buttonAlert: {
    ...theme.classes.buttonAlert,
    bottom: 0,
    marginTop: 10,
  },

  buttonLabel: {
    ...theme.classes.buttonLabel,
  },

  containerStyle: {
    width: '100%',
    borderBottomColor: theme.colors.accent3,
    borderBottomWidth: 1,
    backgroundColor: theme.values.defaultListLinkBackgroundColor,
  },

  subTitle: {
    color: theme.colors.accent4,
    ...theme.fonts.body,
  },

  title: {
    ...theme.fonts.title2,
  },
});

export default AboutScreen;
