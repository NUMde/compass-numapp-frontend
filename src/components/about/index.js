// components for the 'about' screen
export { default as AboutListItem } from './aboutListItem';
export { default as AboutListLink } from './aboutListLink';
export { default as RedirectModal } from './redirectModal';
