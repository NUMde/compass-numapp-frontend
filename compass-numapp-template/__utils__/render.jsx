import React from 'react';
import { Provider } from 'react-redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react-native';
import { buildStore } from './mockStore';

let store = null;

export function renderWithRedux(subComponent, state = {}) {
  if (!store) {
    store = buildStore(state);
  }

  const queries = render(<Provider store={store}>{subComponent}</Provider>);

  return {
    ...queries,
    store,
  };
}

export function initRenderModule(initialState = {}) {
  store = buildStore(initialState);

  return store;
}
