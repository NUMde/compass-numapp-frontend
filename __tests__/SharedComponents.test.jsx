// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

/***********************************************************************************************
imports
***********************************************************************************************/

import React from 'react';
import { renderWithRedux } from '../__utils__/render';

import { RedirectModal } from '../src/components/about';
import QuestionnaireModal from '../src/components/questionnaireModal';
import {
  Banner,
  Spinner,
  ScrollIndicatorWrapper,
} from '../src/components/shared';
import ProgressBar from '../src/components/questionnaireModal/progressbar';

/***********************************************************************************************
tests
***********************************************************************************************/

// these are mostly render tests and can be extended any time ;D

describe('SHARED COMPONENTS:', () => {
  // simple render test
  test('<Banner /> can be rendered', () => {
    // renders the component
    const tree = renderWithRedux(<Banner />);

    // checks if the component matches the snapshot
    expect(tree).toMatchSnapshot();
  });

  // simple render test
  test('<Progressbar /> can be rendered', () => {
    // renders the component
    const tree = renderWithRedux(<ProgressBar />);

    // checks if the component matches the snapshot
    expect(tree).toMatchSnapshot();
  });

  // simple render test
  test('<RedirectModal /> can be rendered', () => {
    // renders the component with mocked props
    const tree = renderWithRedux(
      <RedirectModal
        actions={{ hideModal: () => jest.fn() }}
        modalLink={{ title: 'test-string' }}
      />,
    );

    // checks if the component matches the snapshot
    expect(tree).toMatchSnapshot();
  });

  // simple render test
  test('<QuestionnaireModal /> can be rendered', () => {
    // renders the component
    const tree = renderWithRedux(<QuestionnaireModal />);

    // checks if the component matches the snapshot
    expect(tree).toMatchSnapshot();
  });

  // simple render test
  test('<ScrollIndicatorWrapper /> can be rendered', () => {
    // renders the component
    const tree = renderWithRedux(<ScrollIndicatorWrapper />);

    // checks if the component matches the snapshot
    expect(tree).toMatchSnapshot();
  });

  // simple render test
  test('<Spinner /> can be rendered', () => {
    // renders the component
    const tree = renderWithRedux(<Spinner />);

    // checks if the component matches the snapshot
    expect(tree).toMatchSnapshot();
  });
});
