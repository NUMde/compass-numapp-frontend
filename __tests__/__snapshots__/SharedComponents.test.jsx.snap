// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHARED COMPONENTS: <Banner /> can be rendered 1`] = `
<View
  style={
    Object {
      "alignContent": "stretch",
      "alignItems": "stretch",
      "backgroundColor": "#F8F9FA",
      "flexDirection": "column",
      "height": 354.06976744186045,
      "justifyContent": "flex-start",
      "overflow": "hidden",
      "paddingTop": 30,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
        "paddingTop": 10,
        "width": "100%",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
          "marginLeft": 15,
          "marginRight": 15,
          "textAlign": "center",
        }
      }
    >
      <Text
        accessibilityRole="header"
        numberOfLines={1}
        style={
          Object {
            "alignContent": "center",
            "alignSelf": "center",
            "color": "#1F2532",
            "flex": 1,
            "fontFamily": "IBMPlexSans-SemiBold",
            "fontSize": 33.75,
            "lineHeight": 43.125,
            "textAlign": "center",
            "width": "100%",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 3,
            },
            undefined,
            false,
          ]
        }
      >
        <View
          accessibilityHint="Opens new page with additional informations and links."
          accessibilityLabel="Additional Information"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 44,
              "opacity": 1,
              "width": 104.65116279069767,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#EB576A",
                          "fontSize": 48.83720930232558,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    Û∞çú
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        accessibilityRole="header"
        numberOfLines={1}
        style={
          Object {
            "alignSelf": "center",
            "color": "#1F2532",
            "fontFamily": "IBMPlexSans",
            "fontSize": 30,
            "lineHeight": 39.375,
            "marginBottom": 8.720930232558139,
            "marginTop": -26.162790697674417,
            "position": "relative",
            "textAlign": "center",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-end",
        "left": 0,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 354.06976744186045,
          "width": 750,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-end",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "bottom": 18.313953488372093,
          "maxHeight": 432.29448350459705,
          "maxWidth": 650,
          "position": "relative",
        }
      }
    />
  </View>
</View>
`;

exports[`SHARED COMPONENTS: <Progressbar /> can be rendered 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#DADADA",
      "borderRadius": 2.5,
      "height": 5,
      "marginHorizontal": 5,
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1F2532",
        "borderRadius": 2.5,
        "height": 5,
        "width": "NaN%",
      }
    }
  />
</View>
`;

exports[`SHARED COMPONENTS: <QuestionnaireModal /> can be rendered 1`] = `<View />`;

exports[`SHARED COMPONENTS: <RedirectModal /> can be rendered 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  onSwipeComplete={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={50}
  scrollTo={[Function]}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "borderColor": "#F8F9FA",
            "borderRadius": 10,
            "justifyContent": "flex-end",
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 80,
          },
          Object {
            "margin": 0,
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onSwipeComplete={[Function]}
      panResponderThreshold={4}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={50}
      scrollTo={[Function]}
      statusBarTranslucent={false}
      style={
        Object {
          "borderColor": "#F8F9FA",
          "borderRadius": 10,
          "flex": 1,
          "justifyContent": "flex-end",
          "margin": 37.5,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 80,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "backgroundColor": "#F8F9FA",
            "borderColor": "#F8F9FA",
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "height": "auto",
            "maxHeight": "90%",
            "paddingLeft": 20,
            "paddingRight": 20,
          }
        }
      >
        <RCTScrollView
          onScroll={[Function]}
          scrollEventThrottle={16}
        >
          <View>
            <View
              style={
                Object {
                  "paddingBottom": 20,
                  "paddingTop": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderBottomColor": "#F8F9FA",
                    "borderBottomWidth": 1,
                    "marginBottom": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#1F2532",
                      "fontFamily": "IBMPlexSans-SemiBold",
                      "fontSize": 45,
                      "lineHeight": 54.375,
                      "marginBottom": 5,
                    }
                  }
                >
                  test-string
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "color": "#1F2532",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 28.125,
                    "lineHeight": 37.5,
                  }
                }
              />
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#DADADA",
            "height": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F8F9FA",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "height": "auto",
              "margin": 10,
              "width": 44,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 3,
              },
              undefined,
              false,
            ]
          }
        >
          <View
            accessibilityHint="Switches to default browser application, and opens link there."
            accessibilityLabel="accept"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 28,
                    "height": 52,
                    "margin": 7,
                    "overflow": "hidden",
                    "width": 52,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#1F2532",
                        "borderRadius": 28,
                        "height": 52,
                        "justifyContent": "center",
                        "width": 52,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#ffffff",
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      Û∞Ñ¨
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": "auto",
              "margin": 10,
              "width": 44,
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`SHARED COMPONENTS: <ScrollIndicatorWrapper /> can be rendered 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    enableOnAndroid={true}
    innerRef={[Function]}
    onScroll={[Function]}
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`SHARED COMPONENTS: <Spinner /> can be rendered 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "#1F2532",
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#EB576A"
        hidesWhenStopped={true}
        size="large"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</Modal>
`;
